File: 0648-replace-words/0648-replace-words.py
class Solution:
    def replaceWords(self, dictionary: List[str], sentence: str) -> str:
        result = ''
        dictionary.sort(key=len)
        words = sentence.split(' ')

        for word in words:
            sub_str = word
            for item in dictionary:
                if sub_str.startswith(item):
                    sub_str = item
                    break
            result += sub_str + ' '

        return result.strip()

        

File: 0648-replace-words/0648-replace-words.js
/**
 * @param {string[]} dictionary
 * @param {string} sentence
 * @return {string}
 */
var replaceWords = function(dictionary, sentence) {
  dictionary.sort((a, b) => a.length - b.length);
  let result = '';
  let words = sentence.split(' ');

  for (let word of words) {
    let sub_str = word;
    for (let item of dictionary) {
      if (sub_str.startsWith(item)) {
        sub_str = item;
        break;
      }
    }
    result += sub_str + ' ';
  }

  return result.trim();
};